Rebus/Workers/ThreadBased/ThreadWorker.cs

```diff
@@ -183,14 +187,18 @@ void DoWork(bool onlyRunContinuations = false)
             }
         }
 
-        async Task<ReceiveResult> TryReceiveTransportMessage(DefaultTransactionContext transactionContext)
+        async Task<ReceiveResult> TryReceiveTransportMessage(DefaultTransactionContext transactionContext, CancellationToken cToken)
         {
             try
             {
-                var message = await _transport.Receive(transactionContext);
+                var message = await _transport.Receive(transactionContext, cToken);
 
                 return new ReceiveResult(message);
             }
+            //catch (TaskCanceledException tex)
+            //{
+            //    return new ReceiveResult(tex, true);
+            //}
```

Comment:
fair enough 😁  I'm on vacation right now and I will be back on Thursday next week.... I cannot promise that I will merge your PR before then...I hope that's ok 😄